src/Features/Core/Portable/NavigationBar/IRemoteNavigationBarItemService.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Remote;
+
+namespace Microsoft.CodeAnalysis.NavigationBar
+{
+    internal interface IRemoteNavigationBarItemService
+    {
+        ValueTask<ImmutableArray<RoslynNavigationBarItem>> GetItemsAsync(
+            PinnedSolutionInfo solutionInfo, DocumentId documentId, bool supportsCodeGeneration, CancellationToken cancellationToken);
```

Comment:
standard OOP rpc pattern.  Note that we can just return RoslynNavigationBarItem as it's a simple marshallable type.